picard/ui/scripteditor.py

```diff
@@ -0,0 +1,549 @@
+
+        # Add scripting documentation to parent frame.
+        doc_widget = ScriptingDocumentationWidget(self)
+        self.ui.documentation_frame_layout.addWidget(doc_widget)
+        self.ui.show_documentation.stateChanged.connect(self.toggle_documentation)
+
+        self.ui.file_naming_format.textChanged.connect(self.check_formats)
+        self.ui.file_naming_word_wrap.stateChanged.connect(self.toggle_wordwrap)
+        self.ui.import_script.clicked.connect(self.import_script)
+        self.ui.export_script.clicked.connect(self.export_script)
+
+        self.ui.example_filename_sample_files_button.clicked.connect(self.update_example_files)
+        self._sampled_example_files = []
+
+        self.ui.example_filename_after.itemSelectionChanged.connect(self.match_before_to_after)
+        self.ui.example_filename_before.itemSelectionChanged.connect(self.match_after_to_before)
+
+        self.ui.preset_naming_scripts.clear()
+        for item in PRESET_SCRIPTS:
+            self.ui.preset_naming_scripts.addItem(item["title"])
+        self.ui.preset_naming_scripts.setCurrentIndex(0)
+        self.ui.preset_naming_scripts.currentIndexChanged.connect(self.select_script)
+
+        def sync_vertical_scrollbars(widgets):
+            """Sync position of vertical scrollbars for listed widgets.
+            """
+            def _sync_scrollbar_vert(widget, value):
+                widget.blockSignals(True)
+                widget.verticalScrollBar().setValue(value)
+                widget.blockSignals(False)
+
+            widgets = set(widgets)
+            for widget in widgets:
+                for other in widgets - {widget}:
+                    widget.verticalScrollBar().valueChanged.connect(
+                        partial(_sync_scrollbar_vert, other))
+
+        # Sync example lists vertical scrolling
+        sync_vertical_scrollbars((self.ui.example_filename_before, self.ui.example_filename_after))
+
+        # Set highlight colors for selected list items
+        example_style = self.ui.example_filename_before.palette()
+        highlight_bg = example_style.color(QPalette.Active, QPalette.Highlight)
+        highlight_fg = example_style.color(QPalette.Active, QPalette.HighlightedText)
+        stylesheet = "QListView::item:selected { color: " + highlight_fg.name() + "; background-color: " + highlight_bg.name() + "; }"
+        self.ui.example_filename_after.setStyleSheet(stylesheet)
+        self.ui.example_filename_before.setStyleSheet(stylesheet)
+
+        # Set splitter background color to highlight background color to make splitters more visible (disabled because it looks ugly)
+        # stylesheet = "QSplitter::handle { background: " + highlight_bg.name() + "; }"
+        # self.ui.splitter_between_editor_and_examples.setStyleSheet(stylesheet)
+        # self.ui.splitter_between_editor_and_documentation.setStyleSheet(stylesheet)
+        # self.ui.splitter_between_before_and_after.setStyleSheet(stylesheet)
```

Comment:
If code is commented out, better remove it from source (but keep it in commit history)